File: Client/stock_tkl/lib/models/realtime_quote.dart
class RealtimeQuote {
  final String symbol;
  final String companyName;
  final String industry;
  final String sector;
  final double price;
  final double changeValue;
  final double percentChange;
  final int volume;

  RealtimeQuote(
      {required this.symbol,
      required this.companyName,
      required this.industry,
      required this.sector,
      required this.price,
      required this.changeValue,
      required this.percentChange,
      required this.volume});

  factory RealtimeQuote.fromJson(Map<String, dynamic> json) => RealtimeQuote(
      symbol: json['symbol'],
      companyName: json['companyName'],
      industry: json['industry'],
      sector: json['sector'],
      price: json['price'],
      changeValue: json['changeValue'],
      percentChange: json['percentChange'],
      volume: json['volume']);

  static List<RealtimeQuote> listFromJson(list) =>
      List<RealtimeQuote>.from(list.map((x) => RealtimeQuote.fromJson(x)));
}

File: Server/StockTKL/src/main/java/com/example/stocktkl/model/RealtimeQuote.java
package com.example.stocktkl.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.math.BigDecimal;

/**
 * @author Tan Dang
 * @since 17/11/2023 - 10:54 pm
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
public class RealtimeQuote {
    private String symbol;
    private String companyName;
    private String industry;
    private String sector;
    private BigDecimal price;
    private BigDecimal changeValue;
    private BigDecimal percentChange;
    private int volume;
}
